### ISSUE TITLE
Thumbnail URLs wrongly detected as text/xml

### ISSUE THREAD

**Issue by martin9000andersen** [xbmc/xbmc#15187]
### Describe the bug
Here is a clear and concise description of what the problem is:

When using the add-on plugin.video.drnu a lot of the thumbnails are missing.

The logfile shows that Kodi thinks that the URLs for the thumbnails have mimeType "text/xml".
I have manually opened all the ULRs and all of theme are automatically redirected to a different url and shows the image with out any problem.
Some of the thumbnails are downloaded corectly by Kodi, for these a original url and a "effective URL" are printed to log. The mimType are logged as "image/jpeg".

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

## Possible Fix

I have tried to debug the C++ code. In the file TextureCacheJob.cpp in the function CTextureCacheJob::LoadImage(), there is a check for "file.IsPicture()" and the function returns NULL e.g. when mimeType "text/xml". If I uncomment the "return NULL;" all thumbnails are shown correctly in Kodi.

I do not know why the two lines:
    file = CFileItem(image, false);
    file.FillInMimeType();
think most of the URLs are "text/url", when "CBaseTexture::LoadFromFile(image,...)" always downloads an image.

### To Reproduce

1. Install add-on: DR TV (Sorry this add-on might only work in Denmark)
2. Open add-on DR TV
3. select "DR tv" -> "Programs A-Z" -> "A"
4. You will now see that a lot (possible all) thumbnails are missing.

### Debuglog

The debuglog can be found here:
 https://paste.kodi.tv/amorovufel.kodi

### Screenshots 
Here are some links or screenshots to help explain the problem:

In this screenshot there are missing thumbnails
![kodi_thumbnails_missing](https://user-images.githubusercontent.com/2692211/50612208-7c714180-0ed9-11e9-9735-876adb81a4a7.jpeg)

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [x] Linux
 - [ ] OSX
 - [ ] Raspberri-Pi
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name:
 - Kodi version: 18-rc4 (also tested on 17 with same result)
(I have tested on LibreELEC 9-beta running on Odroid C2, and on a standard PC with Kodi running in Ubuntu 18.04)

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by lrusak** [comment]
This most likely has to do with the redirect. I'm not sure how we handle it with libcurl but from command line I see this.

```
$ curl -I https://www.dr.dk/mu-online/api/1.2/bar/53cd1126a11f9d13c8629841
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0HTTP/2 302
location: https://asset.dr.dk/ImageScaler/?file=/mu-online/api/1.2/asset/53cd1126a11f9d13c862983f%2525253Fraw=True&w=940&h=529&scaleAfter=crop&quality=75
server: Microsoft-IIS/8.5
x-aspnet-version: 4.0.30319
x-powered-by: ASP.NET
access-control-allow-origin: *
content-length: 0
x-cacheable: YES:max-age=300
x-varnish: 282365868
x-dr-appname: dir_1001287
x-via: varnishwwwol02.lx.dmz.dr.dk
x-edgeconnect-midmile-rtt: 0
x-edgeconnect-origin-mex-latency: 151
x-edgeconnect-midmile-rtt: 69
x-edgeconnect-origin-mex-latency: 151
cache-control: public, max-age=170
date: Wed, 02 Jan 2019 21:08:23 GMT
strict-transport-security: max-age=15768000 ; preload
```

vs

```

$ curl -IL https://www.dr.dk/mu-online/api/1.2/bar/53cd1126a11f9d13c8629841
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0  124k    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0HTTP/2 302
location: https://asset.dr.dk/ImageScaler/?file=/mu-online/api/1.2/asset/53cd1126a11f9d13c862983f%2525253Fraw=True&w=940&h=529&scaleAfter=crop&quality=75
server: Microsoft-IIS/8.5
x-aspnet-version: 4.0.30319
x-powered-by: ASP.NET
access-control-allow-origin: *
content-length: 0
x-cacheable: YES:max-age=300
x-varnish: 282365868
x-dr-appname: dir_1001287
x-via: varnishwwwol02.lx.dmz.dr.dk
x-edgeconnect-midmile-rtt: 0
x-edgeconnect-origin-mex-latency: 151
x-edgeconnect-midmile-rtt: 69
x-edgeconnect-origin-mex-latency: 151
cache-control: public, max-age=113
date: Wed, 02 Jan 2019 21:09:20 GMT
strict-transport-security: max-age=15768000 ; preload

HTTP/2 200
content-type: image/jpeg
server: Microsoft-IIS/10.0
x-via: Imagescol02
fritz: 14
content-length: 127724
cache-control: public, max-age=14823
expires: Thu, 03 Jan 2019 01:16:24 GMT
date: Wed, 02 Jan 2019 21:09:21 GMT
strict-transport-security: max-age=15768000
```

**Comment by lrusak** [comment]
can you try this patch? https://github.com/lrusak/xbmc/commit/97afcebeb2588cce3ec3e9a1fb293bb68a6adc61

**Comment by martin9000andersen** [comment]
Thank you very much for the patch, unfortunately it did not make a difference.

I think you are right when you assume it is the redirect that are the problem.
I have now tried to hack the add-on so all the URL are pointing to the final destination
(Python code:
1:    tmp = requests.get(imageURL, allow_redirects=True, stream=True).url
2:    imageURL = tmp
)
This solved the problem, and all images was always loaded correctly.
I also tried removing line 2, to verify that the problem still existed when the URLs were just loaded by the python add-on. The result of this test was that more images was loaded correctly in Kodi, but not all, so I think that we can conclude that hack with final-destination-URLs solves the problem.

**Comment by vpeter4** [comment]
Kodi/curl already follows redirection. But it seems the server really sends text/xml content as seen from debug log:
```
15:59:30.307 T:140329021843200   DEBUG: Curl::Debug - SSL_DATA_OUT: 
15:59:30.307 T:140329021843200   DEBUG: Curl::Debug - HEADER_OUT: HEAD /ImageScaler/?file=/mu-online/api/1.2/asset/5b96975fa11faa04e8b758ef%2525253Fraw=True&w=940&h=529&scaleAfter=crop&quality=75 HTTP/1.1
15:59:30.307 T:140329021843200   DEBUG: Curl::Debug - HEADER_OUT: Host: asset.dr.dk
15:59:30.307 T:140329021843200   DEBUG: Curl::Debug - HEADER_OUT: User-Agent: Kodi/18.0-RC3 (X11; Linux x86_64) LibreELEC/9.0 App_Bitness/64 Version/18.0-RC3-Git:18.0rc3-Leia
15:59:30.307 T:140329021843200   DEBUG: Curl::Debug - HEADER_OUT: Accept: */*
15:59:30.307 T:140329021843200   DEBUG: Curl::Debug - HEADER_OUT: Referer: https://www.dr.dk/mu-online/api/1.2/bar/5b96975fa11faa04e8b758f3
15:59:30.307 T:140329021843200   DEBUG: Curl::Debug - HEADER_OUT: Accept-Charset: UTF-8,*;q=0.8
15:59:30.329 T:140329021843200   DEBUG: Curl::Debug - SSL_DATA_IN: 
15:59:30.329 T:140329021843200   DEBUG: Curl::Debug - HEADER_IN: HTTP/1.1 200 OK
15:59:30.329 T:140329021843200   DEBUG: Curl::Debug - HEADER_IN: Content-Length: 1172
15:59:30.329 T:140329021843200   DEBUG: Curl::Debug - HEADER_IN: Content-Type: text/xml
15:59:30.329 T:140329021843200   DEBUG: Curl::Debug - HEADER_IN: Last-Modified: Wed, 02 Jan 2019 14:46:50 GMT
15:59:30.329 T:140329021843200   DEBUG: Curl::Debug - HEADER_IN: Accept-Ranges: bytes
15:59:30.329 T:140329021843200   DEBUG: Curl::Debug - HEADER_IN: ETag: "f32da4fea9a2d41:0"
15:59:30.329 T:140329021843200   DEBUG: Curl::Debug - HEADER_IN: Server: Microsoft-IIS/10.0
15:59:30.329 T:140329021843200   DEBUG: Curl::Debug - HEADER_IN: fritz: 14
15:59:30.329 T:140329021843200   DEBUG: Curl::Debug - HEADER_IN: Date: Fri, 04 Jan 2019 14:59:30 GMT
15:59:30.329 T:140329021843200   DEBUG: Curl::Debug - HEADER_IN: Connection: keep-alive
15:59:30.329 T:140329021843200   DEBUG: Curl::Debug - HEADER_IN: Strict-Transport-Security: max-age=15768000
```

Curl from command line works fine as already sen.

**Comment by martin9000andersen** [comment]
I tried the curl command
curl -IL https://www.dr.dk/mu-online/api/1.2/bar/53cd1126a11f9d13c8629841
multiple times, and it was a little bit random if the returned Content-Type was "test/xml" or "image/jpeg".
Then I tried a different url (https://www.dr.dk/mu-online/api/1.2/bar/53cd1126a11f9d13c8629841) and the Content-Type was "text/xml" 25 times in a row. Then I tried to run curl without "-I" one time, after that the curl command with "-IL" started to return "image/jpeg" every time.

I think that the conclusion is that the server should return "image/jpeg" every time, and never "text/xml".

But when I look in the Kodi code, why is unknown ("application/octet-stream") accepted? could "text/xml" be treated like unknown in this case?
